src/backend/distributed/planner/distributed_planner.c

```diff
@@ -238,10 +256,31 @@ distributed_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 						errhint("Consider using PL/pgSQL functions instead.")));
 	}
 
+	if (localReferenceTableJoin)
+	{
+		/* See comments for IsLocalReferenceTableJoinPlan */
+		result->utilityStmt = (Node *) parse;
```

Comment:
This is pretty hacky and ugly way to do this, alternatives are (1) using CustomScan nodes (2) checking the parse tree in the executor again. 1 might be a bit more complicated to implement, 2 might slow down all of the worker queries quite a bit. This is a temporary solution and we will remove this after fixing the limitations of local execution. Let's chat about this more tomorrow.